CMake/PGICompilers.cmake

```diff
@@ -1,3 +1,7 @@
+if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 21.3 AND ENABLE_OMP_TASKLOOP)
```

Comment:
Why CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 21.3 ? Does 21.2 not abort or somehow implement the feature?

A more cmake-like and robust solution would be to test for the feature availability and not test for any specific compiler version. i.e. Test compile a have_omp_taskloop.cpp code, set a configure variable. Then the code will naturally start working when 22.0 comes along, or whichever version supports it. Keep ENABLE_OMP_TASKLOOP for manual testing and workarounds.

Reply:
Good point on testing the feature. Will change.

Reply:
Added a feature test. 21.2 doesn't abort. I don't think the feature is implemented but probably no-op in 21.2. Once there is a bit of tasking (probably due to target nowait) in the runtime, no-op is wrong for taskloop. My workaround is setting taskgroup.